### PR TITLE
Don't ignore platform requirements for PHP 8.4.

### PR DESCRIPTION
**Pull request by ADmad**
All our dependencies should be PHP 8.4 compatible now.

### CODE REVIEW

**Comment by ADmad** [comment]
We need to figure out what to do with `E_STRICT` usage in `PhpError::$levelMap`. It's deprecated in PHP 8.4 and triggers a warning.

**Comment by ADmad** [comment]
> Passing E_USER_ERROR to trigger_error() is deprecated since 8.4, throw an exception or call exit with a string message instead

:disappointed:

File: src/Error/PhpError.php [hunk 1]
```diff
@@ -95,6 +94,10 @@ public function __construct(
         ?int $line = null,
         array $trace = []
     ) {
+        // E_STRICT is deprecated in PHP 8.4 and using it triggers a deprecation warning.
+        // phpcs:ignore
+        $this->levelMap[@E_STRICT] = 'strict';
```
Comment (markstory) [inline src/Error/PhpError.php]: Should we guard this with a PHP version check? We can not have strict in PHP 8.4 if it is going away.

Comment (ADmad) [reply src/Error/PhpError.php]: If we want to just skip using E_STRICT in PHP 8.4 I can wrap this in a version check.

Comment (markstory) [reply src/Error/PhpError.php]: I think that's a reasonable solution. If PHP effectively doesn't have E_STRICT in 8.4 we don't need to have support for ti.

Comment (ADmad) [reply src/Error/PhpError.php]: Done, only the issue of using `E_USER_ERROR` with `trigger_error()` remains now.

**Comment by ADmad** [comment]
@markstory Would changing `E_USER_ERROR` to `E_USER_WARNING` here achieve the same result?

https://github.com/cakephp/cakephp/blob/ff7df0c75b91049ae3f5db70c035a738f401d081/src/Error/ExceptionTrap.php#L362-L378

**Comment by markstory** [comment]
> Would changing E_USER_ERROR to E_USER_WARNING here achieve the same result?

I think so if `E_USER_ERROR` is going away then the next best thing we have is a warning.

**Pull request merged**